breaking-change(scanner): WPScan works on C shell (#1299)

FuCrowRabbit opened the pull request:
# What did you implement:

I want to work WPScan on Sakura's Rental Server. (*)
WPScan works on C shell.
Changed `2>/dev/null` to `--quiet`.

Fixes #1299 (-1)

(*) ... Not work completely this PR yet.

## Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

https://github.com/FuCrowRabbit/test-vuls-proposal-sakura-rental-server

## Before

```
Error: Too many positional arguments: 2
```

## After

### Theme

```json
[{"name":"twentynineteen","status":"inactive","update":"none","version":"2.1"},{"name":"twentytwenty","status":"inactive","update":"none","version":"1.8"},{"name":"twentytwentyone","status":"active","update":"none","version":"1.4"}]
```

### Plugin

```json
[{"name":"akismet","status":"inactive","update":"none","version":"4.1.12"},{"name":"hello","status":"inactive","update":"none","version":"1.7.2"}]
```

# Checklist:

- [ ] Write tests
- [ ] Write documentation
- [x] Check that there aren't other open pull requests for the same issue/feature
- [x] Format your source code by `make fmt`
- [x] Pass the test by `make test`
- [ ] Provide verification config / commands
- [x] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

***Is this ready for review?:*** YES  

# Reference

* https://make.wordpress.org/cli/handbook/references/config/

FuCrowRabbit commented:
日本語で失礼いたします。
このP/Rに対しての作業は、既に最初のコミットのみで完了しています。
何かこちらで忘れている事がありましたら、大変お手数ですがご教示いただけますでしょうか？
よろしくお願いいたします。

kotakanbe commented:
Hi, @FuCrowRabbit 

Sorry for the late reply.
I have checked, but this fix cannot be merged because the Warning message makes it impossible to parse. 
Therefore, we have to solve this problem in some other way.
Any ideas?


```
[kusanagi@kusanagi]$ wp theme list --path=$PWD/DocumentRoot --quiet --format=json
PHP Warning:  "continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"? in /home/kusanagi/DocumentRoot/wp-content/plugins/unyson/framework/extensions/backup/includes/lib/srdb.class.php on line 854
Warning: "continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"? in /home/kusanagi/DocumentRoot/wp-content/plugins/unyson/framework/extensions/backup/includes/lib/srdb.class.php on line 854
[{"name":"dentist_wp","status":"active","update":"none","version":"1.6.6"},{"name":"twentyfifteen","status":"inactive","update":"available","version":"2.0"},{"name":"twentynineteen","status":"inactive","update":"available","version":"1.2"},{"name":"twentyseventeen","status":"inactive","update":"available","version":"1.6"},{"name":"twentysixteen","status":"inactive","update":"available","version":"1.5"},{"name":"twentytwenty","status":"inactive","update":"available","version":"1.1"},{"name":"twentytwentyone","status":"inactive","update":"none","version":"1.4"}]
```


```
[kusanagi@kusanagi2 ]$ wp theme list --path=$PWD/DocumentRoot --format=json 2>/dev/null
[{"name":"dentist_wp","status":"active","update":"none","version":"1.6.6"},{"name":"twentyfifteen","status":"inactive","update":"available","version":"2.0"},{"name":"twentynineteen","status":"inactive","update":"available","version":"1.2"},{"name":"twentyseventeen","status":"inactive","update":"available","version":"1.6"},{"name":"twentysixteen","status":"inactive","update":"available","version":"1.5"},{"name":"twentytwenty","status":"inactive","update":"available","version":"1.1"},{"name":"twentytwentyone","status":
```

FuCrowRabbit commented:
Hi, @kotakanbe 

I have an idea.
`wp theme list --format=json --exec="error_reporting(E_ERROR|E_PARSE);"`

I haven't tested it yet because I couldn't get the poor plugin/theme.

FuCrowRabbit commented:
Hi, @kotakanbe 

> I haven't tested it yet because I couldn't get the poor plugin/theme.

I was able to reproduce the problem by running Wordpress 4.9 on PHP 7.4.

```bash
wp plugin list --format=json --path=./public/ --allow-root
```

```
[22-Nov-2021 11:01:40 UTC] PHP Warning:  "continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"? in /var/www/html/public/wp-includes/pomo/plural-forms.php on line 210
Warning: "continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"? in /var/www/html/public/wp-includes/pomo/plural-forms.php on line 210
[22-Nov-2021 11:01:41 UTC] PHP Deprecated:  Function get_magic_quotes_gpc() is deprecated in /var/www/html/public/wp-includes/load.php on line 651
Deprecated: Function get_magic_quotes_gpc() is deprecated in /var/www/html/public/wp-includes/load.php on line 651
[{"name":"akismet","status":"inactive","update":"none","version":"4.0.1"},{"name":"hello","status":"inactive","update":"available","version":"1.6"},{"name":"wp-multibyte-patch","status":"inactive","update":"none","version":"2.8.1"}]
```

Then, `--exec="error_reporting(E_ERROR|E_PARSE);"` method did not work.


My next idea is as follows.

```sh
 ( wp plugin list --format=json [...options] > /dev/tty ) >& /dev/null
```

example...

```sh
bash-5.1# ( wp plugin list --format=json --path=./public/ --allow-root > /dev/tty ) >& /dev/null
[{"name":"akismet","status":"inactive","update":"none","version":"4.0.1"},{"name":"hello","status":"inactive","update":"available","version":"1.6"},{"name":"wp-multibyte-patch","status":"inactive","update":"none","version":"2.8.1"}]
bash-5.1# csh
# ( wp plugin list --format=json --path=./public/ --allow-root > /dev/tty ) >& /dev/null
[{"name":"akismet","status":"inactive","update":"none","version":"4.0.1"},{"name":"hello","status":"inactive","update":"available","version":"1.6"},{"name":"wp-multibyte-patch","status":"inactive","update":"none","version":"2.8.1"}]
```

FuCrowRabbit commented:
Changed from "Breaking-Change" to "Fix" as it was determined that compatibility would be maintained.

I also rewrote the test again.
https://github.com/FuCrowRabbit/test-vuls-proposal-sakura-rental-server
